io.github.theforceengine.tfe.yml

```diff
@@ -0,0 +1,68 @@
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
+  # Audio
+  - --socket=pulseaudio
+  # Access to the default Steam Dark Forces data directory for native and Flatpak installations
+  - --filesystem=~/.local/share/Steam/steamapps/common/Dark Forces/Game/
+  - --filesystem=~/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Dark Forces/Game/
+
+modules:
+  - shared-modules/glu/glu-9.json
+  - shared-modules/glew/glew.json
+
+  - name: tfe
+    sources:
+      - type: git
+        url: https://github.com/luciusDXL/TheForceEngine.git
+        tag: v1.09.540
+        x-checker-data:
+          type: git
+          tag-pattern: ^v([\\d.]+)$
+      # Separate engine data from user data on Linux by setting PATH_USER_DOCUMENTS to XDG_HOME.
+      - type: patch
+        path: user_data.patch
+      # Patch automatic detection of Dark Forces installation to only check default Steam path
+      - type: patch
+        path: game_data_detection.patch
+      # Include local metainfo.xml file until a TFE release is made that includes it
+      # https://github.com/luciusDXL/TheForceEngine/pull/375
+      - type: file
+        path: io.github.theforceengine.tfe.metainfo.xml
+      # Add a wrapper script that points TFE_DATA_HOME to the correct location within the sandbox
+      - type: script
+        dest-filename: io.github.theforceengine.tfe.sh
+        commands:
+          - # Check for game data
+          - if [[ -z "$TFE_SKIP_DATACHECK" ]] && [[ -z $(find "$XDG_DATA_HOME" -iname dark.gob) ]] && [[ ! -d "$HOME/.local/share/Steam/steamapps/common/Dark Forces/Game/" ]] && [[ ! -d "/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Dark Forces/Game/" ]]; then
+          -     zenity --error --text "<b>Could not find Dark Forces game data</b>\n\n
+                    Please either install Dark Forces via Steam to your default library path
+                    or copy the game data to <tt><b>$XDG_DATA_HOME/</b></tt>."
+                    --ok-label "Close" --width=400
+          - fi
+          - export TFE_DATA_HOME=/app/share/TheForceEngine
+          - exec theforceengine
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DDISABLE_SYSMIDI=ON
+    builddir: true
+    build-commands:
+      # Install launcher script
+      - install -Dm 744 ../io.github.theforceengine.tfe.sh -t /app/bin
+      # Edit and install desktop launcher file
+      - desktop-file-edit --set-key=Exec --set-value=io.github.theforceengine.tfe.sh ../TheForceEngine/TheForceEngine.desktop
+      # Install locally-included appstream metadata file until a TFE release is made that includes it
+      # https://github.com/luciusDXL/TheForceEngine/pull/375
+      - install -Dm 644 ../io.github.theforceengine.tfe.metainfo.xml /app/share/metainfo/io.github.theforceengine.tfe.metainfo.xml
+      # - install -Dm 644 /run/build/tfe/TheForceEngine/TheForceEngine.metainfo.xml /app/share/metainfo/io.github.theforceengine.tfe.metainfo.xml
```

Comment:
```suggestion
```

Reply:
Ah, does flatpak-builder automatically detect the metainfo files with that filename?

Reply:
this line is a comment, you don't need it. there is no detection, it's installed by the previous line.

Reply:
Yeah, I realised that as soon as I wrote my reply :D the commented-out line is what it'll want to be once TFE merge the metainfo and tag a release but I'm sure I'll be able to remember that when the time comes.

Reply:
it should be done by the build system. this is not specific to flatpak.

(`/app` is just the prefix)

Reply:
Will add that to their cmakelists in my PR, then!